src/datasources/tag/TagDataSource.ts

```diff
@@ -73,18 +78,43 @@ export class TagDataSource extends DataSourceBase<TagQuery> {
 
       return result
     } else {
-      const history = await this.getTagHistoryValues(tag.path, tag.workspace ?? tag.workspace_id, range, maxDataPoints);
-      result.fields = [
-        { name: 'time', values: history.datetimes },
-        { name, values: history.values },
-      ];
-    }
+      const tagPropertiesMap: Record<string, Record<string, string> | null> = {};
+      tagsWithValues.forEach((tag: TagWithValue) => {
+        tagPropertiesMap[tag.tag.path] = tag.tag.properties
+      });
+      const workspaceFromFirstTag = tagsWithValues[0].tag.workspace ?? tagsWithValues[0].tag.workspace_id;
+      const tagHistoryResponse = await this.getTagHistoryWithChunks(
+        Object.keys(tagPropertiesMap),
+        workspaceFromFirstTag,
```

Comment:
So when visualizing the current value of tags, the matched tags can be from a mixed set of workspaces, but for historical values, it's locked to the workspace of the first matched tag. I think this could be confusing for users. We may need to force users to explicitly pick a workspace or make separate requests for each unique workspace.

@TJ-G

Reply:
I can group tags by workspace, change **getTagHistoryWithChunks** signature and as you said, make separate requests for each workspace.

So, in case we have tags with the same path (or displayName), should I prefix the workspace name to the display names ​​so that they can be differentiated?

@mure @TJ-G @joshuaprewitt

Reply:
Where is this displayed to the user? Are we talking about in the panel legend?
<img width="1000" alt="image" src="https://github.com/ni/systemlink-grafana-plugins/assets/7433612/ce3d9a3b-0dd4-4e38-b6cf-6213d56eb89e">

If so, yeah a format like {workspace}.{tagpath} could be a good option.

Reply:
@TigranVardanyan are there any additional PRs for AAU open?  If not and this is the only one, are we complete once this is merged?

Reply:
@markacamps there are 2 more features that I would like to add, but they are not required for AAU

Reply:
@markacamps @TigranVardanyan - This conversation is unresolved and looks like it might require a response to TJ's question.

Reply:
not required for AAU, we can close this one